connector/azure-storage/src/main/java/it/pagopa/selfcare/mscore/connector/azure_storage/AzureBlobClient.java

```diff
@@ -95,6 +94,36 @@ public String getTemplateFile(String templateName) {
         }
     }
 
+    @Override
+    public File getFileAsPdf(String contractTemplate) throws IOException {
+        log.info("START - getFileAsPdf for template: {}", contractTemplate);
+        BlobInputStream blobInputStream = null;
+        FileOutputStream fileOutputStream = null;
+        try{
+            final CloudBlobContainer blobContainer = blobClient.getContainerReference(azureStorageConfig.getContainer());
+            final CloudBlockBlob blob = blobContainer.getBlockBlobReference(contractTemplate);
+            blobInputStream = blob.openInputStream();
+            String fileName = Paths.get(contractTemplate).getFileName().toString();
+            File downloadedFile = File.createTempFile(fileName, ".pdf");
+            fileOutputStream = new FileOutputStream(downloadedFile);
+            byte[] buffer = new byte[4096];
+            int bytesRead;
+            while ((bytesRead = blobInputStream.read(buffer)) != -1) {
+                fileOutputStream.write(buffer, 0, bytesRead);
+            }
+
+            log.info("END - getFileAsPdf");
+            return downloadedFile;
+        } catch (StorageException | URISyntaxException | IOException e) {
+            log.error(String.format(ERROR_DURING_DOWNLOAD_FILE.getMessage(), contractTemplate), e);
+            throw new MsCoreException(String.format(ERROR_DURING_DOWNLOAD_FILE.getMessage(), contractTemplate),
+                    ERROR_DURING_DOWNLOAD_FILE.getCode());
+        } finally {
+            // Close the streams
+            blobInputStream.close();
+            fileOutputStream.close();
```

Comment:
if `blobInputStream.close()` throws `fileOutputStream.close()` won't be executed, it would be better to handle this using try-with-resources statement.

Reply:
try something like this, close action will perform by try-catch:

```
public File getFileAsPdf(String contractTemplate){
        log.info("START - getFileAsPdf for template: {}", contractTemplate);

        final CloudBlobContainer blobContainer;
        final CloudBlockBlob blob;
        final File downloadedFile;
        
        try {
            blobContainer = blobClient.getContainerReference(azureStorageConfig.getContainer());
            blob = blobContainer.getBlockBlobReference(contractTemplate);

            String fileName = Paths.get(contractTemplate).getFileName().toString();
            downloadedFile = File.createTempFile(fileName, ".pdf");
        } catch (URISyntaxException | StorageException | IOException e) {
            log.error(String.format(ERROR_DURING_DOWNLOAD_FILE.getMessage(), contractTemplate), e);
            throw new MsCoreException(String.format(ERROR_DURING_DOWNLOAD_FILE.getMessage(), contractTemplate),
                    ERROR_DURING_DOWNLOAD_FILE.getCode());
        }

        try (BlobInputStream blobInputStream = blob.openInputStream();
             FileOutputStream fileOutputStream = new FileOutputStream(downloadedFile)){
            
            byte[] buffer = new byte[4096];
            int bytesRead;
            while ((bytesRead = blobInputStream.read(buffer)) != -1) {
                fileOutputStream.write(buffer, 0, bytesRead);
            }

        } catch (IOException | StorageException e) {
            log.error(String.format(ERROR_DURING_DOWNLOAD_FILE.getMessage(), contractTemplate), e);
            throw new MsCoreException(String.format(ERROR_DURING_DOWNLOAD_FILE.getMessage(), contractTemplate),
                    ERROR_DURING_DOWNLOAD_FILE.getCode());
        }

        log.info("END - getFileAsPdf");
        return downloadedFile;
    }
```